### ISSUE TITLE
P7-ML-002: XGBoost Multi-Label Training Pipeline

### ISSUE THREAD

**Issue by flexycode** [Artificial-Ledger-Technology/ALT-Flex#226]
### P7-ML-002: XGBoost Multi-Label Training Pipeline

**Title**: Train and Validate the One-vs-Rest XGBoost Classifier

| Field           | Value                              |
| --------------- | ---------------------------------- |
| Priority        | P0 — Critical                      |
| Estimated Hours | 12                                 |
| Dependencies    | P7-ML-001                          |
| Assigned Agent  | `senior_machine_learning_engineer` |
| QA Agent        | `senior_qa_engineer`               |
| Review Agent    | `senior_code_reviewer`             |
| Labels          | `ml`, `xgboost`, `training`        |

**Description**:
Train a multi-label XGBoost classifier using the One-vs-Rest strategy across the 10 exploit pattern categories. Validate using stratified k-fold cross-validation and produce per-pattern metrics.

**Acceptance Criteria**:

- [ ] Training script `scripts/ml/train_model.py` that:
  - Loads `exploit_features.csv`
  - Applies One-vs-Rest wrapper around `XGBClassifier`
  - Uses stratified 5-fold cross-validation
  - Reports macro-averaged Precision, Recall, F1 per fold
- [ ] Target: **Macro F1 ≥ 0.80** across 10 categories
- [ ] Per-pattern confusion matrix exported as markdown (thesis appendix)
- [ ] Threshold sensitivity analysis at 9 thresholds (0.10–0.90)
- [ ] Feature importance plot exported as PNG for thesis Chapter 4/5
- [ ] Random Forest baseline comparison included for thesis defense
- [ ] Trained model saved as:
  - `research/models/xgboost_exploit_classifier.json` (XGBoost native)
  - `research/models/xgboost_exploit_classifier.onnx` (ONNX export)

**Key Configuration**:

```python
# XGBoost Hyperparameters (initial)
xgb_params = {
    "objective": "binary:logistic",  # per-label binary
    "eval_metric": "logloss",
    "max_depth": 6,
    "learning_rate": 0.1,
    "n_estimators": 200,
    "subsample": 0.8,
    "colsample_bytree": 0.8,
    "scale_pos_weight": "auto",  # handle class imbalance
    "random_state": 42,
}
```

---
